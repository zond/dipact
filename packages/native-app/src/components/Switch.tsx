import React from "react";
import { StyleProp, Switch as RNSwitch, Text, View } from "react-native";
import { useTheme } from "../hooks";

const useStyles = (): StyleProp<any> => {
	const theme = useTheme();
	return {
		switchContainer: {
			display: "flex",
			flexDirection: "row",
			alignItems: "center",
		},
		switch: {
			display: "flex",
			alignSelf: "flex-start",
		},
	};
};

interface SwitchProps {
	value: boolean;
	onValueChange: (value: boolean) => void;
	label?: string;
}

const Switch = ({ value, onValueChange, label }: SwitchProps) => {
	const theme = useTheme();
	const styles = useStyles();

	return (
		<View style={styles.switchContainer}>
			<RNSwitch
				testID="switch"
				style={styles.switch}
				trackColor={{
					true: theme.palette.secondary.light,
					false: theme.palette.background.dark,
				}}
				thumbColor={
					value ? theme.palette.secondary.main : theme.palette.paper.main
				}
				onValueChange={onValueChange}
				value={value}
			/>
			<Text>{label}</Text>
		</View>
	);
};

export default Switch;
