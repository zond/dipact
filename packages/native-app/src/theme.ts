import { FullTheme } from "react-native-elements";

const gridSize = 8;

const palette = {
	primary: {
		light: "#63ccff",
		main: "#281A1A",
		dark: "#006db3",
		contrastText: "#FDE2B5",
	},
	secondary: {
		light: "#fef1da",
		main: "#FDE2B5",
		dark: "#006db3",
		contrastText: "#000",
	},
	paper: {
		main: "#FFF"
	},
	background: {
		main: "#FFF",
		dark: "#AAA"
	},
	border: {
		main: "#000",
	},
	notification: {
		main: "#000",
	},
	text: {
		main: "#000",
		light: "#57606a",
	}
};

export const theme: Partial<FullTheme> = {
  colors: {
	  primary: palette.primary.main,
	  secondary: palette.secondary.main,
  },
  palette,
  spacing: (value: number) => value * gridSize,
}