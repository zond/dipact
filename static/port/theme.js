

const theme = MaterialUI.createMuiTheme({ 
  typography: {
    useNextVariants: true,
    fontFamily: [
      'Cabin',
      'Roboto',
      'sans-serif',
    ].join(','),
  },
  palette: {
    primary: {
      light: '#63ccff',
      main: '#281A1A',
      dark: '#006db3',
      contrastText: '#FDE2B5',
    },
     secondary: {
      light: '#63ccff',
      main: '#FDE2B5',
      dark: '#006db3',
      contrastText: '#000',
    },
  },
 }); 
 
 export default theme;
