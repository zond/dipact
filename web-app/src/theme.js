// See https://stackoverflow.com/questions/61220424/material-ui-drawer-finddomnode-is-deprecated-in-strictmode
import { unstable_createMuiStrictModeTheme as createMuiTheme } from '@material-ui/core/styles';


const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
    fontFamily: ["Cabin", "Roboto", "sans-serif"].join(","),
  },
  palette: {
    primary: {
      light: "#63ccff",
      main: "#281A1A",
      dark: "#006db3",
      contrastText: "#FDE2B5",
    },
    secondary: {
      light: "#63ccff",
      main: "#FDE2B5",
      dark: "#006db3",
      contrastText: "#000",
    },
    success: {
      main: "#008000",
    },
  },
});

export default theme;
