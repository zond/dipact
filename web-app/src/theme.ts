// See https://stackoverflow.com/questions/61220424/material-ui-drawer-finddomnode-is-deprecated-in-strictmode
import { unstable_createMuiStrictModeTheme as createMuiTheme } from "@mui/material/styles";
// it could be your App.tsx file or theme file that is included in your tsconfig.json
import { Theme } from "@mui/material/styles";

declare module "@mui/styles/defaultTheme" {
  interface DefaultTheme extends Theme {}
}

const theme = createMuiTheme({
  typography: {
    fontFamily: ["Cabin", "Roboto", "sans-serif"].join(","),
  },
  palette: {
    primary: {
      light: "#63ccff",
      main: "#281A1A",
      dark: "#006db3",
      contrastText: "#FDE2B5",
    },
    secondary: {
      light: "#63ccff",
      main: "#FDE2B5",
      dark: "#006db3",
      contrastText: "#000",
    },
    success: {
      main: "#008000",
    },
  },
});

export default theme;
