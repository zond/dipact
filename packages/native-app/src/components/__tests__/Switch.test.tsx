import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "react-native-elements";
import { fireEvent, render, waitFor } from "@testing-library/react-native";

import Switch from "../Switch";
import { theme } from "../../theme";

const WrappedSwitch = (props: Parameters<typeof Switch>[0]) => {
	return (
		<ThemeProvider theme={theme}>
			<Switch {...props} />
		</ThemeProvider>
	);
};

describe("Snapshot tests", () => {
	let onValueChange = () => {};
	test("False without label", () => {
		const props = { value: false, onValueChange };
		const tree = renderer.create(<WrappedSwitch {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	test("False with label", () => {
		const props = { value: false, label: "Click here", onValueChange };
		const tree = renderer.create(<WrappedSwitch {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	test("True without label", () => {
		const props = { value: true, onValueChange };
		const tree = renderer.create(<WrappedSwitch {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	test("True with label", () => {
		const props = { value: true, label: "Click here", onValueChange };
		const tree = renderer.create(<WrappedSwitch {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe("Functional tests", () => {
	let props: Parameters<typeof Switch>[0];
	beforeEach(() => {
		props = { value: false, onValueChange: jest.fn() };
	})
	test("Click when false", () => {
		const { getByTestId } = render(<WrappedSwitch {...props} />);
		const _switch = getByTestId("switch");
		fireEvent(_switch, "valueChange", true);
		expect(props.onValueChange).toBeCalledWith(true);
	});
	test("Click when true", () => {
		const { getByTestId } = render(<WrappedSwitch {...props} />);
		const _switch = getByTestId("switch");
		fireEvent(_switch, "valueChange", false);
		expect(props.onValueChange).toBeCalledWith(false);
	});
	test("Shows label", () => {
		props.label = "Switch label";
		const { getByText } = render(<WrappedSwitch {...props} />);
		getByText(props.label);
	});
});
