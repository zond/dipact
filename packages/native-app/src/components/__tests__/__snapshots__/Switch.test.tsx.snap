// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests False with label 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#fef1da"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "alignSelf": "flex-start",
          "display": "flex",
        },
      ]
    }
    testID="switch"
    thumbTintColor="#FFF"
    tintColor="#AAA"
    value={false}
  />
  <Text>
    Click here
  </Text>
</View>
`;

exports[`Snapshot tests False without label 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#fef1da"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "alignSelf": "flex-start",
          "display": "flex",
        },
      ]
    }
    testID="switch"
    thumbTintColor="#FFF"
    tintColor="#AAA"
    value={false}
  />
  <Text />
</View>
`;

exports[`Snapshot tests True with label 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#fef1da"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "alignSelf": "flex-start",
          "display": "flex",
        },
      ]
    }
    testID="switch"
    thumbTintColor="#FDE2B5"
    tintColor="#AAA"
    value={true}
  />
  <Text>
    Click here
  </Text>
</View>
`;

exports[`Snapshot tests True without label 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#fef1da"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "alignSelf": "flex-start",
          "display": "flex",
        },
      ]
    }
    testID="switch"
    thumbTintColor="#FDE2B5"
    tintColor="#AAA"
    value={true}
  />
  <Text />
</View>
`;
